ably/realtime_conn_test.go

```diff
@@ -0,0 +1,74 @@
+package ably_test
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ably/ably-go/ably"
+	"github.com/ably/ably-go/ably/testutil"
+)
+
+func record(t *testing.T, spy map[int]int, done func() bool) (chan<- ably.State, func()) {
+	ch := make(chan ably.State, 64)
+	stop := time.After(timeout)
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-stop:
+				t.Fatalf("record timed out after %v (spy=%v)", timeout, spy)
+			case s := <-ch:
+				spy[s.State]++
```

Comment:
I know it's quite a simple test so I would be surprised if it happens here, but shouldn't this be an array instead of a `map` where you would have the history of what happened instead of just how many time the connection was in this state?